src/conmon.c

```diff
@@ -728,6 +701,90 @@ static gboolean conn_sock_cb(int fd, GIOCondition condition, gpointer user_data)
 	return G_SOURCE_REMOVE;
 }
 
+static gboolean conn_sock_cb(G_GNUC_UNUSED int fd, GIOCondition condition, gpointer user_data);
+
+static void sock_try_write_to_masterfd_stdin(struct conn_sock_s *sock)
+{
+	if (!sock->remaining)
+		return;
+
+	ssize_t w = write(masterfd_stdin, sock->buf + sock->off, sock->remaining);
+	if (w < 0) {
+		nwarn("Failed to write to container stdin");
+	} else {
+		sock->off += w;
+		sock->remaining -= w;
+	}
+}
+
+static void write_to_masterfd_stdin(gpointer data, gpointer user_data)
+{
+	struct conn_sock_s *sock = (struct conn_sock_s *)data;
+	bool *has_data = user_data;
+
+	sock_try_write_to_masterfd_stdin(sock);
+
+	if (sock->remaining)
+		*has_data = TRUE;
```

Comment:
why `TRUE` here when it's cast as a bool? either work, but you use `false` below

Reply:
thanks for noticing.

 We should use true\false for `bool` but TRUE/FALSE for `gboolean`.